redis/src/types.rs

```diff
@@ -121,9 +120,23 @@ pub enum Value {
     /// to express nested structures.
     Bulk(Vec<Value>),
     /// A status response.
-    Status(String),
+    Status(String), // maybe change to SimpleString ?
     /// A status response which represents the string "OK".
     Okay,
+    /// Ordered map value from the server. use `as_map_iter` function.
+    Map(Vec<Value>),
+    /// Unordered set value from the server.
+    Set(Vec<Value>),
+    /// A floating number response from the server.
+    Double(f64),
+    /// A boolean response from the server.
+    Boolean(bool),
+    /// First String is format and other is the string
+    VerbatimString(String, String),
+    /// Very large number that out of the range of the signed 64 bit numbers
+    BigNumber(String),
+    /// Push data from the server.
+    Push(Vec<Value>),
```

Comment:
I'm trying to verify this, but it seems like the first value in the push data can be an enum too - the current wording implies that the only valid types are PubSub and Monitor. Not sure whether we want to use this in the type or not.

Reply:
If not an enum, at least separate it as a string. Consider adding field names, for clarity.

Reply:
Sure, i think using enum for push type is better

Reply:
I discussed it with a maintainer of OSS Redis, and they said that this shouldn't be an enum, but `VerbatimString` can use an enum. The first field is likely to be used by modules, so we can't limit it to a closed set of values. 
IMO the type should be `Push(String, Vec<Value>)`